aiosupervisor/models/addons.py

```diff
@@ -13,7 +13,11 @@
 
 
 class AddonStage(StrEnum):
-    """AddonStage type."""
+    """AddonStage type.
+
+    Supervisor may add new AddonStages in future as a non-breaking change.
+    Those should be returned as string in response models until next update.
```

Comment:
Doesn't this mean that clients can't trust this model?

Reply:
We'd only would add new stages. So you can trust that the currently existing three will be there.

It depends a bit on the use case of a string enum if a library user can make safe assumptions/implement it in a forward compatible way. E.g. the `HostFeature` enum is quite straight forward. New features will be added, but a library user should just not care.

This is definitely a bit more controversial: Currently we have `experimental`, `stable`, `deprecated`, and the reflect the life cycle in that order. If we'd introduce `beta` which presumably is between `experimental` and `stable`, a client library would most likely handle this wrong :cry: 

For the library user, it probably would be better if commit to the current stages and keep this one fixed :thinking: .

We do have add-on channel on the road map, so that ties into stages. These will be `stable`, `beta` and `development`, and will most likely more important then the add-on stage really.

IMHO we should stick with the three stages we have, so make fields using `AddonStage` just taking `AddonStage`.

Reply:
No that wasn't the intention. The situation we're dealing with here is that Supervisor wants to be able to add new values to these some of fields with a fixed set of values in the future. But the problem is this:
1. Supervisor auto-updates
2. Supervisor must be updated before everything else (core, addons, etc.)

That means these enums basically can never change without breaking all existing clients, not even additive changes (it's ok that removing a value breaks clients as Supervisor should never do that).

When talking it over with Stefan what we settled on was for enums supervisor wants to be able to make additive changes to we note that somewhere. And then when a field uses that enum the type of the field is `AddonStage | str`. That way if Supervisor adds a new `AddonStage` existing clients won't break, it'll just be a string.

We do not expect users to try and handle string values though. They should just write a case switch based on the enum. And then the next time the supervisor client updates itll incorporate that new value into the enum and they can update their case case switch.

Reply:
> Adding new add-on stages is definitely a bit more controversial: Currently we have experimental, stable, deprecated, and the reflect the life cycle in that order. If we'd introduce something new, e.g. say beta, which presumably would be between experimental and stable, a client library would most likely handle this wrong 😢

I'm not understanding why this one would be controversial. To my knowledge no current client of supervisor uses this information anywhere (including core and frontend). It's effectively just metadata we have available that no one really adopted. Making it a breaking change requiring a new version of the API to add a new stage seems very heavy handed.

What kind of use cases are you thinking of @agners where someone would use this where a new stage would break them?